core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -321,17 +326,25 @@ protected boolean shouldUpdateStreams(long currentBlockId) {
 
   /**
    * Closes or cancels {@link #mCurrentCacheStream}.
-   *
-   * @throws IOException if the close or cancel fails
    */
-  private void closeOrCancelCacheStream() throws IOException {
+  private void closeOrCancelCacheStream() {
     if (mCurrentCacheStream == null) {
       return;
     }
-    if (mCurrentCacheStream.remaining() == 0) {
-      mCurrentCacheStream.close();
-    } else {
-      mCurrentCacheStream.cancel();
+    try {
+      if (mCurrentCacheStream.remaining() == 0) {
+        mCurrentCacheStream.close();
+      } else {
+        mCurrentCacheStream.cancel();
+      }
+    } catch (IOException e) {
+      if (e.getCause() instanceof BlockDoesNotExistException) {
+        LOG.info("Block {} does not exist when being cancelled.", getCurrentBlockId());
+      } else if (e.getCause() instanceof InvalidWorkerStateException) {
+        LOG.info("Block {} has invalid worker state when being cancelled.", getCurrentBlockId());
+      } else {
+        LOG.info("Close or cancel throws IOExecption {}.", e.getMessage());
```

Comment:
I changed this to LOG.warn now.

I don't think we should throw an exception here because read should continue though cache fails (e.g. timeout because of slow network, server unreachable because of alluxio worker is busy etc).  What do you think?